web-common/src/Data/Json/JsonNTuple.purs

```diff
@@ -0,0 +1,60 @@
+-- This module exposes a data type that is equal to the Data.Tuple type. But
+-- only the minimium methods for encoding and decoding are provided, as their only
+-- use case is to serve as a serialization mechanism with Haskell tuple type.
+module Data.Json.JsonNTuple where
+
+import Prelude
+import Data.Array as Array
+import Data.List (List(..))
+import Data.List as List
+import Data.Tuple (Tuple(..))
+import Foreign (ForeignError(..), fail, readArray)
+import Foreign.Class (class Decode, class Encode, decode, encode)
+
+data JsonNTuple a b
+  = JsonNTuple a b
+
+infixr 6 JsonNTuple as /\
+
+infixr 6 type JsonNTuple as /\
+
+instance showTuple :: (Show a, Show b) => Show (JsonNTuple a b) where
+  show (JsonNTuple a b) = show a <> " /\\ " <> show b
```

Comment:
It might be helpful to surround this string in parentheses

```purescript
"(" <> show a <> " /\\ " <> show b <> ")"
```

Reply:
I thought of it, and it's useful for tuples, but not that much for triples or 4-tuples, etc. I can provide an instance chain but seems an overkill.

I tried to see if I was able to make it recursive, but I couldn't :(.